packages/react-instantsearch-hooks/src/components/__tests__/InstantSearch.test.tsx

```diff
@@ -452,9 +455,10 @@ describe('InstantSearch', () => {
   });
 
   // This test shows that giving an unstable `onStateChange` reference (or any
-  // unstable prop) remounts the <InstantSearch> component and therefore resets
-  // the state after the remount.
-  // Users need to provide stable references for rerenders to keep the state.
+  // unstable prop) does not remount the <InstantSearch> component and therefore
+  // keeps the state.
+  // Prior to the current implementation, we created a new InstantSearch.js
+  // instance at each prop change, which made this test fail.
   test('recovers the state on rerender with an unstable onStateChange', async () => {
```

Comment:
We can remove this test if you think it doesn't make sense to have it now.

Reply:
it makes sense to keep for when the "update" method moves to InstantSearch.js